src/main/kotlin/no/nav/faktureringskomponenten/service/FakturaserieService.kt

```diff
@@ -179,23 +171,21 @@ class FakturaserieService(
         }.referanse
     }
 
-    /**
-     *  Hvis forrige fakturaserie var SINGEL(årsavregning) så tar vi krediteringFakturaRef som den igjen hentet fra sin
-     *  forrige fakturaserie. Denne må være en positiv faktura, ellers fungerer ikke OEBS. Siden alle årsavregninger potensielt
-     *  kan være negative så må vi opprinnelig hente fra fakturaserie som hører til et trygdeavgiftsvedtak.
-     */
     private fun hentKrediteringFakturaRef(
-        fakturaserie: Fakturaserie,
-        fakturaDto: FakturaDto
+        tidligereFakturaserie: Fakturaserie?,
     ): String {
-        val krediteringFakturaRef = if (fakturaserie.intervall == FakturaserieIntervall.SINGEL) {
-            fakturaserie.faktura.single().krediteringFakturaRef
-        } else {
-            val faktura = fakturaserie.faktura.filter { it.overlapperMedÅr(fakturaDto.startDato.year) }
-                .minBy { faktura: Faktura -> faktura.fakturaLinje.first().periodeFra }
-            faktura.hentFørstePositiveFaktura().referanseNr
+        if (tidligereFakturaserie == null) {
+            return ""
         }
-        return krediteringFakturaRef
+
+        require(
+            tidligereFakturaserie.status !in setOf(
+                FakturaserieStatus.ERSTATTET,
+                FakturaserieStatus.KANSELLERT
+            )
+        ) { "Tidligere fakturaserie med ref ${tidligereFakturaserie.referanse} er i feil status: ${tidligereFakturaserie.status}" }
+
```

Comment:
Bør man kansellere tidligere fakturaserie hvis forrige faktura for årsavregningen ikke ble bestilt ennå?

Reply:
Nei, det blir vel feil. Andregangs årsavregning tar utgangspunkt i at den første er fakturert. Dette er en generell svakthet ved løsningen om årsavregning. Vi sjekker hva beregning i api mener skal faktureres, men vi sjekker ikke mot faktiske faktura i faktureringskomponent når vi gjør årsavregningsvedtak